### PR TITLE
Improved `bool_validator` and tests for it

### PR DESCRIPTION
**Pull request by pixel365**
## Change Summary

Minor improvement of the `bool_validator` and tests for it.
Added two new values, and also added trimming of empty string characters.

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by pixel365** [comment]
please review

File: tests/test_validators.py [hunk 1]
```diff
@@ -1256,3 +1256,35 @@ def validate_foo(cls, v):
     with pytest.raises(RuntimeError, match='test error'):
         model.foo = 'raise_exception'
     assert model.foo == 'foo'
+
+
+def test_bool_validator():
```
Comment (PrettyWood) [inline tests/test_validators.py:1261]: A lot of those tests are already in parametrized test `test_types.py::test_default_validators`
You can remove all of this and simply add some usecases there.

**Review by PrettyWood — COMMENTED** [review]
I would really like an open issue or discussion for the rationale behind this.
I've never seen `+` and `-` in any config environment nor have I seen anyone else ask for this.
I'd rather have you customize your own pydantic settings if noone else wants this :/

**Comment by samuelcolvin** [comment]
I don't agree with this:
* `+/-` - they're clear as part of this list, but imagine `allow_negative='-'` I would think that was null or some custom meaning, not false
* the strip is understandable, but it would significantly slow down pydantic-core which currently matches pydantic logic

Thanks for your contribution, sorry it's taken so long to response.

**Pull request closed without merging**